pkg/lib/scoped/syncer.go

```diff
@@ -0,0 +1,116 @@
+package scoped
+
+import (
+	"fmt"
+	"reflect"
+
+	v1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// NewUserDefinedServiceAccountSyncer returns a new instance of UserDefinedServiceAccountSyncer.
+func NewUserDefinedServiceAccountSyncer(logger *logrus.Logger, client operatorclient.ClientInterface, versioned versioned.Interface) *UserDefinedServiceAccountSyncer {
+	return &UserDefinedServiceAccountSyncer{
+		logger:    logger,
+		versioned: versioned,
+		client:    client,
+		clock:     &clock.RealClock{},
+	}
+}
+
+// UserDefinedServiceAccountSyncer syncs an operator group appropriately when
+// a user defined service account is specified.
+type UserDefinedServiceAccountSyncer struct {
+	versioned versioned.Interface
+	client    operatorclient.ClientInterface
+	logger    *logrus.Logger
+	clock     clock.Clock
+}
+
+const (
+	// All logs should in this package should have the following field to make
+	// it easy to comb through logs.
+	logFieldName  = "scoped"
+	logFieldValue = ""
+)
+
+// SyncOperatorGroup takes appropriate actions when a user specifies a service account.
+func (s *UserDefinedServiceAccountSyncer) SyncOperatorGroup(in *v1.OperatorGroup) (out *v1.OperatorGroup, err error) {
+	out = in
+	namespace := in.GetNamespace()
+	serviceAccountName := in.Spec.ServiceAccount
+
+	logger := s.logger.WithFields(logrus.Fields{
+		"operatorGroup": in.GetName(),
+		"namespace":     in.GetNamespace(),
+		logFieldName:    logFieldValue,
+	})
+
+	if serviceAccountName == "" {
+		if in.Status.ServiceAccountRef == nil {
+			return
+		}
+
+		// User must have removed ServiceAccount from the spec. We need to
+		// rest Status to a nil reference.
+		out, err = s.update(in, nil)
+		if err != nil {
+			err = fmt.Errorf("failed to reset status.serviceAccount, sa=%s %v", serviceAccountName, err)
+		}
+		return
+	}
+
+	// A service account has been specified, we need to update the status.
+	sa, err := s.client.KubernetesInterface().CoreV1().ServiceAccounts(namespace).Get(serviceAccountName, metav1.GetOptions{})
+	if err != nil {
+		err = fmt.Errorf("failed to get service account, sa=%s %v", serviceAccountName, err)
+		return
+	}
+
+	// TODO:
+	// 	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/reference"
+	//
+	//  ref, err := reference.GetReference(sa)
+	// 	if err != nil {
+	//	  return
+	//  }
+	// Right now, I am seeing the following error
+	// no kind is registered for the type v1.ServiceAccount in scheme \"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/reference/reference.go:11\
```

Comment:
that referencer only knows about olm types. You should be able to use `client-go/tools/reference` with the scheme for kube types.